Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -141,6 +141,74 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
         //       handled above, thus, if we get to here that mustn't be the case and we should return the empty string.
         return ""_string;
     }
+    case PropertyID::Animation: {
+        auto const get_longhand_as_vector = [&](PropertyID longhand_id) -> StyleValueVector {
+            auto value = longhand(longhand_id);
+
+            if (value->is_value_list())
+                return value->as_value_list().values();
+
+            // FIXME: This is required as parse_comma_separated_value_list() returns a single value directly instead of a list if there's only one.
+            return { value.release_nonnull() };
+        };
+
+        // If we don't have the same number of values for each longhand, we can't serialize this shorthand.
+        if (any_of(m_properties.sub_properties, [&](auto longhand_id) { return get_longhand_as_vector(longhand_id).size() != get_longhand_as_vector(m_properties.sub_properties[0]).size(); }))
+            return ""_string;
+
+        StringBuilder builder;
+        for (size_t i = 0; i < get_longhand_as_vector(m_properties.sub_properties[0]).size(); i++) {
+            auto animation_name = get_longhand_as_vector(PropertyID::AnimationName)[i]->to_string(mode);
+            bool first = true;
+
+            for (auto longhand_id : m_properties.sub_properties) {
+                auto longhand_value = get_longhand_as_vector(longhand_id)[i];
+
+                bool should_serialize_longhand = [&]() {
+                    if (!longhand_value->equals(property_initial_value(longhand_id)))
+                        return true;
+
+                    if (longhand_id == PropertyID::AnimationDuration && !get_longhand_as_vector(PropertyID::AnimationDelay)[i]->equals(property_initial_value(PropertyID::AnimationDelay)))
+                        return true;
+
+                    // https://drafts.csswg.org/css-animations-1/#animation
+                    // Furthermore, when serializing, default values of other properties must be output in at least the
+                    // cases necessary to distinguish an animation-name that could be a value of another property
+                    if (longhand_id == PropertyID::AnimationTimingFunction && first_is_one_of(animation_name, "linear"sv, "ease"sv, "ease-in"sv, "ease-out"sv, "ease-in-out"sv, "step-start"sv, "step-end"sv))
+                        return true;
+
+                    if (longhand_id == PropertyID::AnimationDirection && first_is_one_of(animation_name, "normal"sv, "reverse"sv, "alternate"sv, "alternate-reverse"sv))
```

Comment:
Doesn't need doing in this PR, but: if we generated `string -> enum` functions for each CSS enum, this could check `string_to_animation_direction(animation_name).has_value()` and avoid duplicating the list of possible values.

(I originally commented on the AnimationTimingFunction line, but detecting easing function names is a lot less trivial. :sweat_smile:)